spec/src/main/asciidoc/java-api.adoc

```diff
@@ -100,3 +99,24 @@ public class MyCheck implements HealthCheck {
     }
 }
 ```
+
+As health check procedures are CDI bean they can also be defined with CDI producers:
+
+
+```
+@ApplicationScoped
+class MyChecks {
+
+  @Produces
+  @ApplicationScoped
+  HealthCheck check1() {
+    return () -> return HealthStatus.state(getMemUsage() < 90%);
```

Comment:
@antoinesd I know it's an example but this wouldn't compile as '%' is modulo operator :)